site-scrapers/EastBostonNHC/index.js

```diff
@@ -67,13 +67,18 @@ async function fetchResponse({ url, method, headers, body }) {
         method,
         body,
         headers,
-    }).then(
-        (res) => res.json(),
-        (err) => {
-            console.error(err);
-            return null;
-        }
-    );
+    })
+        .then(
+            (res) => {
+                console.log("logging res from fetch", res);
```

Comment:
Did this work? May have to use:
```javascript
console.log("logging res from fetch", res.text());
```

Reply:
I'd gotten this error: https://github.com/node-fetch/node-fetch/issues/533

Reply:
(but I don't think my current solution works...)

Reply:
you might need `await res.text()`, I hit something today where res.text() was giving me a Promise

Reply:
Sorry, I'm confused. Why does this `.then` have 3 functions? Shouldn't it only have 2?

Reply:
Can also try:
```javascript
.then((res) => res.text())
.then((textRes) => { 
  console.log('Received:', textRes);
  return JSON.parse(textRes);
})
.catch(console.error))
```

Reply:
yeah Liv my code as is was just plain wrong.

Reagan I'll try that.

Reply:
I regret this. I thought to myself, "Seems like the right thing here is to just print the entire http Response object, because we might want that for diagnostics, like if it returns a 404 error or something. I wonder how to do that?"

I thought the answer would be simple, but it appears not to be. Here is an implementation, which is probably fine for here but probably better in common code, and I don't know if it is worth the effort to put in, but since I wasted far too much time figuring it out, I thought I'd offer it here:

```diff
diff --git a/site-scrapers/EastBostonNHC/index.js b/site-scrapers/EastBostonNHC/index.js
index 6126469..de5d6e1 100644
--- a/site-scrapers/EastBostonNHC/index.js
+++ b/site-scrapers/EastBostonNHC/index.js
@@ -62,18 +62,38 @@ async function GetAllAvailability(availabilityService) {
     }, {});
 }
 
+async function dumpResponse(res) {
+    // defined fields from https://developer.mozilla.org/en-US/docs/Web/API/Response
+    const fields = [
+        "ok ",
+        "redirected",
+        "status",
+        "statusText",
+        "type ",
+        "url",
+    ];
+    const obj = {};
+    obj.headers = {};
+    res.headers.forEach((n, v) => (obj.headers[n] = v));
+    fields.forEach((f) => (obj[f] = res[f]));
+    // bye bye async.
+    // obj.text = await res.text();
+    console.log(JSON.stringify(obj));
+}
+
 async function fetchResponse({ url, method, headers, body }) {
     return await fetch(url, {
         method,
         body,
         headers,
     })
-        .then((res) => res.text())
-        .then((textRes) => {
+        .then(async (res) => {
+            let textRes = await res.text();
             try {
                 return JSON.parse(textRes);
             } catch (err) {
                 console.error("Failed to parse response", textRes, err);
+                dumpResponse(res);
                 throw err;
             }
         })
```

The pain points are:
1. You need a fixed list of the fields in a Response to return, there's no way to get them, including not even `Reflection.ownKeys()`
2. The `headers` of a response are an iterator and not an object.
3. You can't get a promise for `res.text()` and still have access to the other parts of `res`, so an async function is used (here an async arrow function!) which waits for `res.text()` before deciding if there is a problem that merits dumping the entire response object.

Use it or not.

Reply:
Thank you for looking into this John! I think it makes sense to keep it simple and see if we can catch issues with what we have, for now.